go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/vmware-tanzu/velero
 
-go 1.22.0
+go 1.22.6
```

Comment:
since this is not a release branch, wouldn't be better to let Go version without patch?

i.e. `1.22`

Reply:
I just followed the existing convention here. 

Personally I prefer to be specific, because this makes the build more reproducible. Also version without patch might use an older cached image for build, however I do not know if this risk applies with Go builds.

Reply:
Suggest not to pin a specific version on the main release. 
If going that way, we need to update the Go version periodically.

Reply:
Just concerned that the binary output then depends on the environment where it is built. For example, the node which builds it may have an older `golang:1.22-bookworm` image cached. And as a result the binary is then built with an older go version.

Would you accept the change if I add Makefile target to update Go version to the current?

Reply:
Does your environment build the Velero image from Dockerfile?
The Velero main docker images are built by the GitHub action triggered by the PR merge.
IMO, they don't have the issue.

We also care about your concerned issue, but the Golang version is pinned on the released versions.

Reply:
I removed patch level version and now the build only refers to **1.22** version